### ISSUE TITLE
little tiny issue, ?syntax err

### ISSUE THREAD

**Issue by pinballelectronica** [Atinoda/text-generation-webui-docker#1]
scripts/build_extensions.sh has windows lf's in it which crashes during build (docker desktop windows) (syntax err)

run dos2unix on it or copy it into notepad++ with the LF's set to Unix and it'll sail through.

Thanks for all the work.

**Comment by pinballelectronica** [comment]
Also fixed this to be relative instead of to root: the entrypoint script didn't have the LF issue.

# Run
COPY scripts/docker-entrypoint.sh scripts/docker-entrypoint.sh
RUN chmod +x scripts/docker-entrypoint.sh
ENTRYPOINT ["scripts/docker-entrypoint.sh"]

**Comment by Atinoda** [comment]
@pinballelectronica thanks for letting me know about the issue, and also testing for a fix! I have modified the .gitattributes and I believe it should fix the issue without needing to use `dos2unix`. Can you please check by fresh clone of the repo and see if it works for you?

The use of an absolute path to the entrypoint script is deliberate. However, do you have a reason why you would suggest to use a relative path instead? I am open to hear your views.

**Comment by pinballelectronica** [comment]
You are right.  Well I have so many ways of running things these days- docker-hyperv or wsl or virtualbox- it's hard to decide which one.  I WANT wsl2 to work but it fails me more often than not.  In this instance I ran docker compose from Windows command line to build.  I git cloned your repo in a directory called i:\something\ so it's relative and therefore was looking for /scripts (root) which was only relative on windows (i:\something\text-generation-webui-docker\scripts\) so that bombed.  

in other words, it doesn't find the docker entrypoint.sh file without removing the ./ and making it relative scripts/docker-entrypoint.sh

excuse my mistakes, english IS my natural language lol..but vodka

**Comment by pinballelectronica** [comment]
Also I'm still having probs getting 7860 listening.  I'm getting an empty response- for the life of me I can't figure it out.  Im thinking somewhere in the code it's listing 127.0.0.1 instead of 0.0.0.0

**Comment by Atinoda** [comment]
All good - I will close this issue as resolved. I've tried WSL2 a bit and coming from Linux I thought it was pretty decent (big improvement on V1, at least!). However, I mostly use Linux these days so I haven't put it through its paces.

That's interesting about the problem with relative paths... I'm not so familiar with running docker on Windows so I will leave the absolute path for now because it is the standard approach. I will bear it mind though, and may revisit later, the project is still taking shape.

*Re: the port 7860 issue - the "--listen" argument should have the application listening on all interfaces. Double-check that `server.py` is launching with the correct arguments and see if that helps. Otherwise, if you're still having problems please open a new issue, and include details of how you've deployed the software :)*

**Comment by pinballelectronica** [comment]
Right.  

127.0.0.1 didn’t send any data.
ERR_EMPTY_RESPONSE

Dunno could be a SO_REUSEADDR problem:

 TCP    0.0.0.0:7860           DESKTOP-178JU918:0      LISTENING
  TCP    0.0.0.0:7860           DESKTOP-178JU918:0      LISTENING

**Comment by pinballelectronica** [comment]
return requests.api.request('get', 'http://127.0.0.1/', **kwargs)
 
 I would say 0.0.0.0 would be a better idea.
